src/simulations/predict_koval.py

```diff
@@ -0,0 +1,54 @@
+import pandas as pd
+
+from sklearn.model_selection import train_test_split
+
+from src.config import INPUTS
+from src.data.read_wfsim import f_w,  time, W_t
+from src.helpers.cross_validation import forward_walk_splitter
+from src.helpers.analysis import fit_statistics
+from src.helpers.features import production_rate_dataset
+from src.helpers.models import load_models, test_model
+from src.models.koval import Koval
+from src.simulations import step_sizes
+
+
+# Loading the previously serialized Koval model
+koval = load_models('koval')['koval']
+
+koval_predictions_file = INPUTS['wfsim']['koval_predictions']
+koval_predictions_metrics_file = INPUTS['wfsim']['koval_predictions_metrics']
+koval_predictions = {
+    'Step size': [], 't_start': [], 't_end': [], 't_i': [], 'Prediction': []
+}
+koval_predictions_metrics = {'Step size': [], 'r2': [], 'MSE': []}
+
+X = W_t[:-1]
+y = f_w[1:]
```

Comment:
There has to be a better way. This is not DRY, put something in the `src/helpers/features.py` file